### PR TITLE
Set battery warning for MAVLink battery status

### PR DESCRIPTION
**Pull request by jonrwat**
**Software Modification**
The software is updated to set the battery warning when a MAVLink BATTERY_STATUS
message is received.  The battery warning is set based on emergency/critical/low
thresholds within the parameters.

This modification is consistent with the existing battery [operation](https://github.com/PX4/Firmware/blob/master/src/modules/systemlib/battery.cpp#L220), when the 'Power Module' is enabled.  A common definition of setting the battery warning could ultimately be achieved (independent of the BAT_SOURCE) with an architectural modification.

**Testing**
Desktop testing performed:
QGroundControl Daily Build (HEAD:e5afeae0)
X-Plane 10.51r2
PX4 Sw: https://github.com/SenteraLLC/Firmware/tree/mavlink_batt_warning-HIL_testing
PX4 Hw: px4fmu-v4 (custom variant)
[Log](https://drive.google.com/open?id=0B-yJurEfkd7eRm1kdFZzX1M3ZTQ)

### CODE REVIEW

**Comment by dagar** [comment]
I initially thought of unifying the systemlib battery usage, but that has filters for voltage, current, and then independently calculates the remaining. Presumably you don't want any of that?

**Comment by jonrwat** [comment]
Correct.  The component sending the MAVLink BATTERY_STATUS message has populated its fields based on a "smart" battery [gas gauge](http://www.ti.com/lsds/ti/power-management/battery-fuel-gauge-overview.page); therefore, no additional filtering or calculations are required.

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 1]
```diff
@@ -1360,6 +1360,26 @@ MavlinkReceiver::handle_message_battery_status(mavlink_message_t *msg)
 	battery_status.cell_count = cell_count;
 	battery_status.connected = true;
 
+	// Get the battery level thresholds.
+	float bat_emergency_thr;
+	float bat_crit_thr;
+	float bat_low_thr;
+	param_get(param_find("BAT_EMERGEN_THR"), &bat_emergency_thr);
```
Comment (dagar) [inline src/modules/mavlink/mavlink_receiver.cpp]: param_find searches the full param set each time. You could do this once and then hold the param_t handle.

**Comment by dagar** [comment]
Looks good. Has this been bench tested with your hardware?

File: src/modules/mavlink/mavlink_receiver.cpp [hunk 2]
```diff
@@ -1360,6 +1363,26 @@ MavlinkReceiver::handle_message_battery_status(mavlink_message_t *msg)
 	battery_status.cell_count = cell_count;
 	battery_status.connected = true;
 
+	// Get the battery level thresholds.
+	float bat_emergen_thr;
+	float bat_crit_thr;
+	float bat_low_thr;
+	param_get(_p_bat_emergen_thr, &bat_emergen_thr);
+	param_get(_p_bat_crit_thr, &bat_crit_thr);
+	param_get(_p_bat_low_thr, &bat_low_thr);
```
Comment (dagar) [inline src/modules/mavlink/mavlink_receiver.cpp]: Ideally we'd still want to only get the parameters when they're updated (parameter uorb message), but for this it should be fine.

Comment (jonrwat) [inline src/modules/mavlink/mavlink_receiver.cpp]: Yes, I bench tested the software by feeding in a MAVLink BATTERY_STATUS message with a varying "battery_status.remaining" value.  The software annunciated the different battery conditions (e.g. Critical Battery) as expected.

**Comment by jonrwat** [comment]
Yes, I bench tested the software by feeding in a MAVLink BATTERY_STATUS message with a varying "battery_status.remaining" value. The software annunciated the different battery conditions (e.g. Critical Battery) as expected.

**Pull request merged**